linera-core/src/worker.rs

```diff
@@ -1244,6 +1244,25 @@ where
                 }
             }
         }
+        if query.request_fallback {
+            if let (Some(epoch), Some(entry)) = (
+                chain.execution_state.system.epoch.get(),
+                chain.unskippable.front().await?,
+            ) {
+                let ownership = chain.execution_state.system.ownership.get();
+                let elapsed = self.storage.current_time().delta_since(entry.seen);
+                if elapsed >= ownership.timeout_config.fallback_duration
+                    && chain.manager.get_mut().vote_fallback(
+                        query.chain_id,
+                        chain.tip_state.get().next_block_height,
+                        *epoch,
+                        self.key_pair(),
+                    )
+                {
```

Comment:
Optional style suggestion: I'm not sure if this helps, but I was wondering if there was a way to make this slightly clearer

```suggestion
                let fallback_allowed = elapsed >= ownership.timeout_config.fallback_duration;
                let mut vote_fallback = || {
                    chain.manager.get_mut().vote_fallback(
                        query.chain_id,
                        chain.tip_state.get().next_block_height,
                        *epoch,
                        self.key_pair(),
                    )
                };
                if fallback_allowed && vote_fallback() {
```

Reply:
Yes, perfect! Thanks :tada: